### PR TITLE
Add vimeo as an external source

### LINKED ISSUE #131: [Feature] Add Vimeo as an  additional External Source
## Problem
This is a sub-issue out of https://github.com/creativecommons/search/issues/48 

## Description
From the additional sources list in the file [get-additional-sources.ts](https://github.com/WordPress/openverse-frontend/blob/main/src/utils/get-additional-sources.ts#L61) that mentioned in https://github.com/creativecommons/search/issues/48, I choose the[Vimeo](https://vimeo.com/search) which provide more of video options free and paid.

Screen Shot 1444-03-25 at 4 47 54 AM

## Additional context
That provide more resource for the Creative Commons engine 

## Implementation

- [x] I would be interested in implementing this feature.
Could you assign me @TimidRobot @possumbilities

### PR DESCRIPTION
**Pull request by Rubadel**
## Fixes

Fixes #131 by @rubadel, @possumbilities assign me 

## Description
Add an additional external search source, vimoe for videos 

## Technical details
I have added all desired implemention code in both search.js and index.php
 
## Tests
1- Define Vimeo in the [engines array] in search.js file.

2- Set it in [ switch case] in [modRights function] in search.js file, 
In that I set license , [rights = &license=by for both] for [ if (comm && deriv)] and [(comm && !deriv)] conditions.

3- set it in [switch case] in [doSearch function] with all needed for done the search, in search.js file, as well as I added this in index.php file within the desired place of the implementation.

## Screenshots

Screen Shot 1444-03-29 at 2 32 01 AM

uncheck commercial use, it without the license

https://user-images.githubusercontent.com/66481905/197650044-d73d22ab-c305-4a68-8812-37a79da9861c.mov

check commercial use  it with the license

https://user-images.githubusercontent.com/66481905/197650011-1ddfe3ca-a20c-45a3-a85b-56d710354894.mov

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dullbenz** [comment]
@Rubadel, You needed not to close the previous pull request (#136) and create a new one after resolving the conflicts. When you resolved the conflicts, you could push to the same branch you were previously working on.
After doing that, you would verify and confirm in your pull request that the conflicts would no longer be present. Then you can tag the reviewer, letting he/she/they know that you have resolved the conflicts.

**Comment by Rubadel** [comment]
@dullbenz Thank you so much for your guideline.
I am new in PR so my knowledge is little but I am learning 🙏🏻

**Comment by Rubadel** [comment]
After last commit 
Screen Shot 1444-03-29 at 4 53 39 AM

**Comment by possumbilities** [comment]
@Rubadel Looks like this is out of date and needs to be updated with the base branch, please do so when you have a moment.

**Comment by Rubadel** [comment]
Updated done

**Comment by possumbilities** [comment]
@Rubadel I believe the logic is a little broken. If one searches for only "use for commercial purposes" they get results that are just a general Vimeo search which is not correct.

**Comment by Rubadel** [comment]
Ok, I will solve it now

**Comment by Rubadel** [comment]
could you see there, this result shows where select for commercial purpose ,, it is licensed by "CC BY" .. is not right?
Screen Shot 1444-04-03 at 12 01 24 AM


And there when unchecked for commercial purpose,, licensed "any"
Screen Shot 1444-04-03 at 12 02 21 AM

**Comment by possumbilities** [comment]
@Rubadel I'm fairly certain that the "any" license choice" is not an umbrella choice, but instead provides all videos, many of which may not be licensed under a CC license at all. 

There are technically three possibilities. 

1. use for commercial purposes; modify, adapt, or build upon.
2. use for commercial purposes;
3. modify, adapt, or build upon.

each then is set as follow: 
1. this defaults to `CC BY` at the moment 
2. this default to `CC BY` at the moment
3. this defaults to `Any License`, which is basically a non-selection and needs to be corrected.

based on the [CC License descriptions](https://creativecommons.org/licenses/) `3` likely needs to be `CC BY-NC`

**Comment by Rubadel** [comment]
Thank you so much for your guid @possumbilities, I resolved that

File: index.php [hunk 1]
```diff
@@ -1,146 +1,383 @@
-<?php
+$(function () {
```
Comment (possumbilities) [inline index.php:1]: You left off the opening `<?php` tag here, please be sure to test any changes you make to ensure that they work prior to pushing them up.

**Review by possumbilities — COMMENTED** [review]
Recent changes mean the codebase is no longer working properly. Please review and address when you have a moment. ❤️

**Comment by Rubadel** [comment]
Hi @possumbilities, firstly I am sorry about these fractions. I solved them in this commit, and there are attach screenshots of the page and of the functionality after my last editing.

**Pic of the page**
Screenshot 1444-04-08 at 11 24 50 PM

**Search with checked commercial and modify, adapt, or build upon**

https://user-images.githubusercontent.com/66481905/199595544-d68cfee3-34e8-4972-aac3-3c07cb195579.mov

**Search with checked commercial only**

https://user-images.githubusercontent.com/66481905/199595735-d57bcbd7-2317-43fe-83e3-2b24d96bd244.mov

**Search with checked modify, adapt, or build upon only**

https://user-images.githubusercontent.com/66481905/199595605-eed1e90c-35fb-4f58-82f6-6edede713eea.mov

**Search with unchecked commercial and modify, adapt, or build upon**

https://user-images.githubusercontent.com/66481905/199595687-972125b9-d625-4f26-94b3-f5ec9cb8127a.mov

**Comment by possumbilities** [comment]
@Rubadel This all looks good, but you spelled Vimeo as `Video`. I believe that's the only remaining change needed now. Please let me know when you've corrected that.

File: index.php [hunk 2]
```diff
@@ -454,14 +468,22 @@ function get_http_security () {
                         </div>
 						<div class="row">
 						<div class="four columns alpha">
+						<div class="engine">
+							<div class="engineButton">
+								<button onclick="setEngine(this)" name="engine" value="vimeo" id="vimeo" aria-label="vimeo"></button>
+							</div>
+							<div class="engineDesc"><label for="vimeo"><strong>Video</strong><br/>Video</label></div>
```
Comment (possumbilities) [inline index.php:475]: `Vimeo` is misspelled as `Video`

Comment (Rubadel) [reply index.php]: @possumbilities I corrected i

**Review by possumbilities — APPROVED** [review]
Tested via Docker, all seems sound. Approved.

**Pull request merged**